src/main/java/com/google/sps/data/PersistentMatchRepository.java

```diff
@@ -0,0 +1,183 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.sps.data.User;
+import java.io.IOException;
+import java.lang.IllegalArgumentException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This class saves User IDs of every User and their matches as Strings.
+ */
+public class PersistentMatchRepository implements MatchRepository {
+  
+  private final DatastoreService datastore;
+  private final Gson gson;
+  private final PersistentUserRepository userRepository;
+  
+  /**
+  * Constructor initializes private variables.
+  */
+  public PersistentMatchRepository() {
+      datastore = DatastoreServiceFactory.getDatastoreService();
+      gson = new Gson();
+      userRepository = new PersistentUserRepository();
+  }
+
+ /**
+  * Private method to add an entity to the datastore, or to override an existing 
+  * entity when new matches are added.
+  */
+  private void addNewEntity(String userId, String matchString) {
+    Entity userEntity = new Entity("User", userId);
```

Comment:
Could you clarify what are the expected properties of the User Entity type? Here you are setting userId and matchIds. The PersistentDataStore sets name, specialties, and company.

Reply:
I renamed it to "UserMatch" to clarify because it is different from the User Entity in the PersistentUserRepository. This entitity stores the User ID and the Collection of that User's match IDs.